ReSharper.FSharp/src/FSharp.Psi.Daemon/src/Stages/FcsErrorsStageProcessBase.fs

```diff
@@ -307,7 +307,7 @@ type FcsErrorsStageProcessBase(fsFile, daemonProcess) =
         | UnitTypeExpected ->
             let expr = nodeSelectionProvider.GetExpressionInRange(fsFile, range, false, null)
             let expr = getResultExpr expr
-            UnitTypeExpectedWarning(expr, error.Message) :> _
+            if isNotNull expr then UnitTypeExpectedWarning(expr, error.Message) else null
```

Comment:
@DedSec256 What do you think about extracting it to something like `createHighlightingFromMappedNode` that would take an additional `mapping: 'T -> 'T` parameter? I can see use doing more things similar to `getResultExpr`, so it'd be great to not have this logic duplicated and to hide the `null` checks inside it.